### PR TITLE
feat: add NodeSelector to PodSpec

### PR DESCRIPTION
**Pull request by shahar-h**
**What type of PR is this?**
Feature - Add support for `NodeSelector` configuration as part of `PodSpec`.

**What this PR does / why we need it**:
It's needed so we can configure a `NodeSelector` for Envoy/RateLimit pods. 
A common use case for `NodeSelector` configuration is running some Envoy/RateLimit pod on a reserved node.

**Which issue(s) this PR fixes**: N/A

### CODE REVIEW

**Comment by shahar-h** [comment]
`coverage-test` failed with race error, which doesn't look related to my change:
```log
--- FAIL: TestProcessHTTPRoutes (0.00s)
    --- FAIL: TestProcessHTTPRoutes/httproute_with_one_filter_from_extension (0.00s)
        testing.go:1465: race detected during execution of test
    testing.go:1465: race detected during execution of test
```

Can someone please re-run?

**Comment by shahar-h** [comment]
Can someone please review?

File: internal/infrastructure/kubernetes/ratelimit/resource_provider.go [hunk 1]
```diff
@@ -183,6 +183,7 @@ func (r *ResourceRender) Deployment() (*appsv1.Deployment, error) {
 					Affinity:                      r.rateLimitDeployment.Pod.Affinity,
 					Tolerations:                   r.rateLimitDeployment.Pod.Tolerations,
 					ImagePullSecrets:              r.rateLimitDeployment.Pod.ImagePullSecrets,
+					NodeSelector:                  r.rateLimitDeployment.Pod.NodeSelector,
```
Comment (zirain) [inline internal/infrastructure/kubernetes/ratelimit/resource_provider.go:186]: can you add a test for this?

Comment (shahar-h) [reply internal/infrastructure/kubernetes/ratelimit/resource_provider.go:186]: done

**Review by zirain — APPROVED** [review]

**Comment by zhaohuabing** [comment]
Should we wait for https://github.com/envoyproxy/gateway/issues/2346 and use the general k8s manifest patch to achieve this?

**Comment by zirain** [comment]
> Should we wait for #2346 and use the general k8s manifest patch to achieve this?

I think this's a common use case, #2346 is design for edge case.

**Review by zhaohuabing — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**